web_src/js/katex.js

```diff
@@ -0,0 +1,41 @@
+import renderMathInElement from 'katex/dist/contrib/auto-render.js';
+
+const mathNodes = document.querySelectorAll('.math');
+
+const ourRender = (nodes) => {
+  for (const element of nodes) {
+    if (element.hasAttribute('katex-rendered') || !element.textContent) {
+      continue;
+    }
+
+    renderMathInElement(element, {
+      delimiters: [
+        {left: '\\[', right: '\\]', display: true},
+        {left: '\\(', right: '\\)', display: false}
+      ],
+      errorCallback: (_, stack) => {
+        element.setAttribute('title', stack);
+      },
+    });
+    element.setAttribute('katex-rendered', 'yes');
+  }
+};
+
+ourRender(mathNodes);
+
+// Options for the observer (which mutations to observe)
+const config = {childList: true, subtree: true};
+
+// Callback function to execute when mutations are observed
+const callback = (records) => {
+  for (const record of records) {
+    const mathNodes = record.target.querySelectorAll('.math');
+    ourRender(mathNodes);
+  }
+};
+
+// Create an observer instance linked to the callback function
+const observer = new MutationObserver(callback);
+
+// Start observing the target node for configured mutations
+observer.observe(document, config);
```

Comment:
Do we really need a `MutationObserver`? A quick check on the GL implementation shows they do without one, and I'd prefer us to do the same as `MutationObserver` is usually rather bad for performance.

Reply:
Unless you want to make KaTeX always available like Mermaid is, we will need to hook into DOM changes for (at least):

* Markdown preview page.
* Loading of comments.

Reply:
`initMarkupContent` already exists and is called in a few places of the code already, just add your rendering function there.

Reply:
I've moved to make math rendering always available in `` ```math `` so I can use those blocks. 

The reason I wasn't hooking in to those was because it was previously completely disabled when math was turned off.